contract/contracts/Officer.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol';
+
+import 'hardhat/console.sol';
+
+interface IBankAccount {
+    function initialize(address tipsAddress, address _sender) external;
+}
+
+interface IVotingContract {
+    function initialize(address _sender) external;
+}
+
+interface IBodContract {
+    function initialize(address[] memory votingAddress) external;
+}
+
+contract Officer is OwnableUpgradeable, ReentrancyGuardUpgradeable, PausableUpgradeable  {
+    
+    address[] founders;
+    address[] members;
+    address bankAccountContract;
+    address votingContract;
+    address bodContract;
+
+    address public bankAccountBeacon;
+    address public employeeContractBeacon;
+    address public votingContractBeacon;
+    address public bodContractBeacon;
+
+    event TeamCreated( address[] founders, address[] members);
+    event ContractDeployed( string contractType, address contractAddress);
+
+  function initialize( address _bankAccountBeacon, address _votingContractBeacon, address _bodContractBeacon) public initializer {
+        __Ownable_init(msg.sender);
+        __ReentrancyGuard_init();
+        __Pausable_init();
+        bankAccountBeacon = _bankAccountBeacon;
+        votingContractBeacon = _votingContractBeacon;
+        bodContractBeacon = _bodContractBeacon;
+    }
+
+   function createTeam(
+    address[] memory _founders, 
+    address[] memory _members
+    ) external  whenNotPaused {
+        require(_founders.length > 0, "Must have at least one founder");
+
+        for (uint256 i = 0; i < _founders.length; i++) {
+            require(_founders[i] != address(0), "Invalid founder address");
+            founders.push(_founders[i]);
+        }
+        for (uint256 i = 0; i < _members.length; i++) {
+            require(_members[i] != address(0), "Invalid member address");
+            members.push(_members[i]);
+        }
+        emit TeamCreated( _founders, _members);
+    }
+
+    function deployBankAccount(address tipsAddress) external onlyOwners whenNotPaused  {
+        require(bankAccountContract == address(0), "Bank account contract already deployed");
+        BeaconProxy proxy = new BeaconProxy(
+            bankAccountBeacon,
+            abi.encodeWithSelector(IBankAccount.initialize.selector, tipsAddress, msg.sender)
```

Comment:
the owner of the bank will be one of the Officer's contract owner? or it's better if we assign all the owners in officer contract to the bank owners?

Reply:
No, Ravi suggested we keep it to one founder for now.